db/memorystore.go

```diff
@@ -116,6 +130,31 @@ func (m *MemoryStore) AddAccount(acct *core.Account) (int, error) {
 	return len(m.accountsByID), nil
 }
 
+func (m *MemoryStore) ChangeAccountKey(acct *core.Account, newKey *jose.JSONWebKey) error {
+	m.Lock()
+	defer m.Unlock()
+
+	oldKeyID, err := keyToID(acct.Key)
+	if err != nil {
+		return err
+	}
+
+	newKeyID, err := keyToID(newKey)
+	if err != nil {
+		return err
+	}
+
+	if otherAccount, present := m.accountsByKeyID[newKeyID]; present {
+		return &ExistingAccountError{"New public key is already in use", otherAccount}
```

Comment:
If you refactor the `ExistingAccountError` as suggested above this will end up just being `return ExistingAccountError{otherAccount}` (I believe you don't need to be using `&{}` here to take a reference)

Reply:
The `&` is needed, otherwise go complains with:
```
db/memorystore.go:148:30: cannot use ExistingAccountError literal (type ExistingAccountError) as type error in return argument:
	ExistingAccountError does not implement error (Error method has pointer receiver)
```
I had to replace
```
func (e *ExistingAccountError) Error() string { ...
```
by
```
func (e ExistingAccountError) Error() string { ...
```
then it worked :)